before applying patch run
sed -i "s/QT5/QT6/g;s/Qt5/Qt6/g" src/cpp/desktop/CMakeLists.txt
--- a/cmake/compiler.cmake
+++ b/cmake/compiler.cmake
@@ -121,7 +121,7 @@ else()
     endif()
   endif() 
 
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
   # Use --as-needed when RSTUDIO_CONFIG_MONITOR_ONLY
   if(RSTUDIO_CONFIG_MONITOR_ONLY)
       foreach(TYPE EXE MODULE SHARED)
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -66,14 +66,14 @@ endif()
 # test directory
 set(TESTS_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/tests/cpp" CACHE STRING "Test includes")
 
-# enable c++11
+# enable c++17
 if(NOT MSVC)
    include(CheckCXXCompilerFlag)
-   CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
-   if(NOT COMPILER_SUPPORTS_CXX11)
-      message(FATAL_ERROR "The compiler ${CMAKE_CXX_COMPILER} has no C++11 support. Please use a different C++ compiler.")
+   CHECK_CXX_COMPILER_FLAG("-std=c++17" COMPILER_SUPPORTS_CXX17)
+   if(NOT COMPILER_SUPPORTS_CXX17)
+      message(FATAL_ERROR "The compiler ${CMAKE_CXX_COMPILER} has no C++17 support. Please use a different C++ compiler.")
    else()
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
    endif()
 endif()
 
--- a/src/cpp/desktop/CMakeLists.txt
+++ b/src/cpp/desktop/CMakeLists.txt
@@ -116,25 +116,27 @@ if(NOT RSTUDIO_ELECTRON)
       -DQT_DEPRECATED_WARNINGS
       -DQT_NO_SIGNALS_SLOTS_KEYWORDS)
    set(CMAKE_INCLUDE_CURRENT_DIR ON)
-
+   #merged qtwebengine
+   #merged(required?) svg,qtbase[xml]
    find_package(Qt6Core REQUIRED)
    find_package(Qt6Widgets REQUIRED)
    find_package(Qt6Gui REQUIRED)
    find_package(Qt6Network REQUIRED)
    find_package(Qt6WebChannel REQUIRED)
-   find_package(Qt6WebEngine REQUIRED)
+   find_package(Qt6WebEngineCore REQUIRED)
    find_package(Qt6WebEngineWidgets REQUIRED)
    find_package(Qt6PrintSupport REQUIRED)
    find_package(Qt6Quick REQUIRED)
    find_package(Qt6QuickWidgets REQUIRED)
-   find_package(Qt6Positioning REQUIRED)
-   find_package(Qt6Sensors REQUIRED)
+   #find_package(Qt6Positioning REQUIRED)
+   #find_package(Qt6Sensors REQUIRED)
    find_package(Qt6Svg REQUIRED)
-   find_package(Qt6Xml REQUIRED)
-   find_package(Qt6XmlPatterns REQUIRED)
+   #find_package(Qt6Xml REQUIRED)
+   #find_package(Qt6XmlPatterns REQUIRED)
    find_package(Qt6Qml REQUIRED)
-   find_package(Qt6Sql REQUIRED)
+   #find_package(Qt6Sql REQUIRED)
    find_package(Qt6OpenGL REQUIRED)
+   find_package(Qt6 REQUIRED COMPONENTS Core5Compat)
    if(UNIX AND NOT APPLE)
       find_package(Qt6DBus REQUIRED)
    endif()
@@ -156,6 +158,7 @@ if(NOT RSTUDIO_ELECTRON)
 
    # include files
    file(GLOB_RECURSE DESKTOP_HEADER_FILES "*.h*")
+   list(FILTER DESKTOP_HEADER_FILES EXCLUDE REGEX ${CMAKE_CURRENT_SOURCE_DIR}/html/.*\\.html)
    if (NOT WIN32)
       list(REMOVE_ITEM DESKTOP_HEADER_FILES ${CMAKE_CURRENT_SOURCE_DIR}/DesktopChooseRHome.hpp)
    endif (NOT WIN32)
@@ -296,14 +299,14 @@ if(NOT RSTUDIO_ELECTRON)
    )
 
    # ensure moc files are generated
-   qt5_wrap_cpp(MOC_DESKTOP_SOURCE_FILES ${MOC_DESKTOP_HEADER_FILES})
-   qt5_wrap_ui(DESKTOP_UI_SOURCES
+   qt6_wrap_cpp(MOC_DESKTOP_SOURCE_FILES ${MOC_DESKTOP_HEADER_FILES})
+   qt6_wrap_ui(DESKTOP_UI_SOURCES
       DesktopChooseRHome.ui
       DesktopInputDialog.ui
    )
 
    set(DESKTOP_RESOURCES_FILES desktop.qrc)
-   qt5_add_resources(DESKTOP_RESOURCES_SOURCES ${DESKTOP_RESOURCES_FILES})
+   qt6_add_resources(DESKTOP_RESOURCES_SOURCES ${DESKTOP_RESOURCES_FILES})
 
    if(WIN32)
 
@@ -346,17 +349,18 @@ if(NOT RSTUDIO_ELECTRON)
       Qt6::Gui
       Qt6::Network
       Qt6::WebChannel
-      Qt6::WebEngine
+      Qt6::WebEngineCore
       Qt6::WebEngineWidgets
       Qt6::PrintSupport
       Qt6::Quick
       Qt6::QuickWidgets
-      Qt6::Positioning
-      Qt6::Sensors
+      #Qt6::Positioning
+      #Qt6::Sensors
       Qt6::Svg
-      Qt6::Xml
-      Qt6::XmlPatterns
+      #Qt6::Xml
+      #Qt6::XmlPatterns
       Qt6::Qml
+      Qt6::Core5Compat
    )
 
    if(UNIX AND NOT APPLE)
@@ -390,7 +394,7 @@ if(NOT RSTUDIO_ELECTRON)
       # These cause issues with macdeployqt due to missing dependencies; suspect we don't need
       # these on any platform.
       set(RSTUDIO_QTMODULES ${RSTUDIO_QTMODULES}
-         Qt6::Sql
+	 #Qt6::Sql
          Qt6::OpenGL
       )
 
@@ -560,7 +564,11 @@ if(NOT RSTUDIO_ELECTRON)
          set(ICU_LIBRARIES libicui18n libicuuc libicudata)
          foreach(ICU_LIBRARY ${ICU_LIBRARIES})
             add_library(${ICU_LIBRARY} SHARED IMPORTED)
-            set_property(TARGET ${ICU_LIBRARY} PROPERTY IMPORTED_LOCATION "${QT_LIBRARY_DIR}/${ICU_LIBRARY}.so")
+           if(EXISTS "${QT_LIBRARY_DIR}64/${ICU_LIBRARY}.so")
+               set_property(TARGET ${ICU_LIBRARY} PROPERTY IMPORTED_LOCATION "${QT_LIBRARY_DIR}64/${ICU_LIBRARY}.so")
+           else()
+               set_property(TARGET ${ICU_LIBRARY} PROPERTY IMPORTED_LOCATION "${QT_LIBRARY_DIR}/${ICU_LIBRARY}.so")
+           endif()
          endforeach()
          target_link_libraries(rstudio ${ICU_LIBRARIES})
 
@@ -570,8 +578,6 @@ if(NOT RSTUDIO_ELECTRON)
             "${CMAKE_CURRENT_BINARY_DIR}/install/LinuxInstallQt.sh"
             @ONLY)
 
-         install_process(
-            "${CMAKE_CURRENT_BINARY_DIR}/install/LinuxInstallQt.sh")
 
       endif()
 
@@ -701,5 +707,6 @@ elseif(WIN32 AND NOT RSTUDIO_SESSION_WIN32)
       ${TESTS_INCLUDE_DIR}
    )
 
+   add_subdirectory(urlopener)
    add_subdirectory(synctex/rsinverse)
 endif()
